From 0216c547fa3cea9ec56f8f5610fdbaab82dc467e Mon Sep 17 00:00:00 2001
From: mj <mj@maxlinear.com>
Date: Sat, 6 Sep 2025 09:28:40 -0700
Subject: [PATCH] WLANRTSYS-92305 don't get second best channel again if second
 best channel is non DFS

Issue: get second best channel after non DFS channel
Fix: don't select second best channel after non DFS channel
---
 src/ap/mxl_dfs_common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ap/mxl_dfs_common.c b/src/ap/mxl_dfs_common.c
index 2db395108..742a39b8d 100644
--- a/src/ap/mxl_dfs_common.c
+++ b/src/ap/mxl_dfs_common.c
@@ -758,6 +758,8 @@ bool mxl_switch_second_best_channel(struct hostapd_iface *iface)
 					return false;
 				}
 			}
+			/* don't get second best channel again if second best channel is non DFS */
+			return true;
 
 		} else {
 			if (mxl_hostapd_zwdfs_switch(iface, HOSTAPD_CAC_STAT_SUCCESS, iface->freq, chwidth, cf_1, cf_2)) {
-- 
2.43.0

